nncf/torch/quantization/quantizer_propagation.py

```diff
@@ -2504,3 +2506,31 @@ def _filter_integer_input_quantizers(self,
             quant_prop_graph.remove_propagating_quantizer(integer_input_pq)
 
         return quant_prop_graph
+
+    def _filter_integer_tensor_quantizers(self, quant_prop_graph: QuantizerPropagationStateGraph) -> \
+            QuantizerPropagationStateGraph:
+        integer_tensor_propagating_quantizers = []
+        filtered_finished_propagating_quantizers = []
+
+        for pq in self._finished_propagating_quantizers:
+            pq_node_key = pq.current_location_node_key
+            pq_node = quant_prop_graph.nodes[pq_node_key]
+            pq_input_edges = quant_prop_graph.in_edges(pq_node)
+            if len(pq_input_edges) == 0:
+                filtered_finished_propagating_quantizers.append(pq)
+                continue
+            assert len(pq_input_edges) == 1
+
+            is_float_node = is_dtype_float(
+                dtype=pq_input_edges[0][InsertionPointGraph.ACTIVATION_DTYPE_EDGE_ATTR]
+            )
+            if is_float_node:
+                filtered_finished_propagating_quantizers.append(pq)
+            else:
+                integer_tensor_propagating_quantizers.append(pq)
+
+        self._finished_propagating_quantizers = filtered_finished_propagating_quantizers
+        for integer_tensor_pq in integer_tensor_propagating_quantizers:
+            quant_prop_graph.remove_propagating_quantizer(integer_tensor_pq)
+
+        return quant_prop_graph
```

Comment:
The correct behaviour would be to use this information in `QuantizerPropagationSolver` and avoid *propagating* the quantizer through integer-typed edges. If we remove the quantizer altogether, we risk that one of the FP32 inputs of the downstream operations will become unquantized. For purposes of stability, however, this solution will have to do.